### PR TITLE
Fixed #24339 -- Added timedelta check and tests for DurationField.prepare_value

### PR DESCRIPTION
**Pull request by orokusaki**
I've run the body of the tests in my shell locally, but haven't run the test suite locally. These commits fix https://code.djangoproject.com/ticket/24339

### CODE REVIEW

File: tests/forms_tests/tests/test_fields.py [hunk 1]
```diff
@@ -642,6 +643,23 @@ class DurationForm(Form):
...
+        field = DurationField()
+        td = datetime.timedelta(minutes=15, seconds=30)
+        self.assertEqual(field.prepare_value(td), duration_string(td))
+
+    def test_durationfield_4(self):
+        """
+        DurationField.prepare_value functions properly when given None.
+        """
+        field = DurationField()
+        self.assertIs(field.prepare_value(None), None)
```
Comment (timgraham) [inline tests/forms_tests/tests/test_fields.py]: You can use `self.assertIsNone` here.

File: tests/forms_tests/tests/test_fields.py [hunk 2]
```diff
@@ -642,6 +643,23 @@ class DurationForm(Form):
             str(f['duration'])
         )
 
+    def test_durationfield_3(self):
+        """
+        DurationField.prepare_value functions properly when given a timedelta.
```
Comment (timgraham) [inline tests/forms_tests/tests/test_fields.py]: I think you can combine the two tests and call it `test_durationfield_prepare_value()`. The test seems self explanatory so I don't think a docstring is needed.

**Comment by orokusaki** [comment]
Thanks for the quick replies @timgraham

**Comment by timgraham** [comment]
Yes, please. You can find the commit message format on the [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist).

**Comment by timgraham** [comment]
buildbot, test this please.

**Comment by timgraham** [comment]
merged in 8a21d250334f56845f255be5534b01d8c6eda314, thanks!

**Comment by orokusaki** [comment]
@timgraham so, an entire line of code was forgotten. I'm not sure if I somehow `CTRL DD` by accident in Vim, or if it was somehow lost during the above process and commits, but see https://github.com/django/django/blob/master/django/forms/fields.py#L516. `prepare_value` only returns a value when `value` is a `timedelta`... while this accidentally works, and the tests pass, it's not technically correct, since the standard is to return the value unchanged when the method can't figure out what to do with it.

To handle this, should I make a ticket on code.djangoproject.com or should I just make a pull request?

**Pull request closed without merging**